MapboxNavigation/NavigationMapView.swift

```
@@ -160,12 +160,23 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
                               contentFrame.minY + contentFrame.height * 0.5))
     }
     
+    open var routeController: RouteController? {
+        didSet {
+            resumeNotifications()
+        }
+    }
+    
     /**
      Determines whether the map should follow the user location and rotate when the course changes.
      - seealso: NavigationMapViewCourseTrackingDelegate
      */
     open var tracksUserCourse: Bool = false {
         didSet {
+            guard let _ = routeController, tracksUserCourse == true else {
+                print("NavigationMapView.routeController must non-nil to set tracksUserCourse to true.")
```

I'd be open to making this an assertion. Fail fast.